### PR TITLE
Add `IntentTEDPolicy` to the default configuration.

### LINKED ISSUE #8415: IntentTEDPolicy: Add policy to default config
**Description of Problem**:
`IntentTEDPolicy ` should become part of the default model configuration so that every new user is automatically using it

**Blockers** (if relevant):
- [ ] `IntentTEDPolicy ` needs to implemented first and be fully functional

**Definition of Done**:
- [ ] the `autoconfig` features automatically includes `IntentTEDPolicy` in the used policies

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- fix #8415 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/shared/importers/default_config.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ pipeline:
 policies:
   - name: MemoizationPolicy
   - name: RulePolicy
-  - name: TEDPolicy
+  - name: IntentTEDPolicy
```
Comment (wochinge) [inline rasa/shared/importers/default_config.yml:30]: Let's keep `TEDPolicy` 🙂

**Review by wochinge — CHANGES_REQUESTED** [review]

**Review by wochinge — CHANGES_REQUESTED** [review]
Now it's missing from `rasa.shared.importers.default_config.yml`, isn't it?

**Review by wochinge — APPROVED** [review]
Did you double check the policy's default params with Daksh/Chris? Looks good to me otherwise 👍🏻

File: data/test_config/config_empty_en_after_dumping.yml [hunk 2]
```diff
@@ -27,6 +27,10 @@ policies:
 # # See https://rasa.com/docs/rasa/policies for more information.
 #   - name: MemoizationPolicy
 #   - name: RulePolicy
+#   - name: IntentTEDPolicy
+#     max_history: 5
+#     epochs: 100
+#     constrain_similarities: true
```
Comment (dakshvar22) [inline data/test_config/config_empty_en_after_dumping.yml:33]: Don't need `constrain_similarities` to be set to `true` for `IntentTEDPolicy`.

Comment (dakshvar22) [reply data/test_config/config_empty_en_after_dumping.yml:33]: In fact, it doesn't even exist in the available configuration parameters.

**Pull request merged**